metpy/plots/skewt.py

```diff
@@ -558,8 +579,12 @@ def plot_mixing_lines(self, w=None, p=None, **kwargs):
         if p is None:
             p = np.linspace(600, max(self.ax.get_ylim())) * units.mbar
 
-        # Assemble data for plotting
+        # Dewpoint contours calculated and converted to xaxis units
         td = dewpoint(vapor_pressure(p, w))
+        if self.ax.xaxis.units is not None:
+            td = td.to(self.ax.xaxis.units)
```

Comment:
Can you use the same call to the current `t0.units` here to convert and remove the conditional here?

Reply:
I will replace the t0 and p0 checks with a call to a (new) function get_axes0 that does ensures the values have units. Unless someone tells me how to force input to have units?

Reply:
Yeah we have decorators to enforce units on the input, so that they would be guaranteed and we wouldn't have to manually assign them via conditionals in this function. I'm just waiting for one of the other maintainers to chime in to see if there's a reason not to do this that I'm missing.